bears/r/FormatRBear.py

```diff
@@ -65,24 +65,21 @@ def create_arguments(filename, file, config_file,
                 1
             ```
         """
-        rcode = ('library(formatR);'
-                 'formatR::tidy_source('
-                 'source="' + escape(filename, '"\\') + '",'
-                 'comment={r_keep_comments},'
-                 'blank={r_keep_blank_lines},'
-                 'arrow={r_use_arrows},'
-                 'brace.newline={r_braces_on_next_line},'
-                 'indent='
-                 '{tab_width}'.format(tab_width=tab_width,
-                                      r_keep_comments=_map_to_r_bool(
-                                          r_keep_comments),
-                                      r_keep_blank_lines=_map_to_r_bool(
-                                          r_keep_blank_lines),
-                                      r_use_arrows=_map_to_r_bool(
-                                          r_use_arrows),
-                                      r_braces_on_next_line=_map_to_r_bool(
-                                          r_braces_on_next_line)))
-        # Disable r_max_expression_length if it is equal to 0
-        rcode += ('' if r_max_expression_length == 0
-                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
+        options = {'source="' + escape(filename, '"\\') + '"': True,
+                   'blank=' + _map_to_r_bool(r_keep_blank_lines):
+                       r_keep_blank_lines is not None,
```

Comment:
the only possiblility for it to be false is to have `r_keep_blank_lines ==
None` that is why i am thinking of removing that, and setting `None` as
default for `r_keep_comments` and `r_use_arrows` :)

2016-06-16 21:18 GMT+01:00 Christian Ngoulla Sob <ngoullasob@gmail.com>:

> isn't it the reverse,
>
> 2016-06-16 20:58 GMT+01:00 Lasse Schuirmann <notifications@github.com>:
>
>> In bears/r/FormatRBear.py
>> <https://github.com/coala-analyzer/coala-bears/pull/525#discussion_r67413460>
>> :
>>
>> > -                 'indent='
>> > -                 '{tab_width}'.format(tab_width=tab_width,
>> > -                                      r_keep_comments=_map_to_r_bool(
>> > -                                          r_keep_comments),
>> > -                                      r_keep_blank_lines=_map_to_r_bool(
>> > -                                          r_keep_blank_lines),
>> > -                                      r_use_arrows=_map_to_r_bool(
>> > -                                          r_use_arrows),
>> > -                                      r_braces_on_next_line=_map_to_r_bool(
>> > -                                          r_braces_on_next_line)))
>> > -        # Disable r_max_expression_length if it is equal to 0
>> > -        rcode += ('' if r_max_expression_length == 0
>> > -                  else ',width.cutoff=' + str(r_max_expression_length)) + ')'
>> > +        options = {'source="' + escape(filename, '"\\') + '"': True,
>> > +                   'blank=' + _map_to_r_bool(r_keep_blank_lines):
>> > +                       r_keep_blank_lines is not None,
>>
>> they all can't be None since they will be a bool right? So this always
>> evaluates to false
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/coala-analyzer/coala-bears/pull/525/files/72dad32bf787c51b55a67dd0146ecbb675de02a4#r67413460>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AJ9xopzyQ9ujJPLAJSc8jSCiSod-tczlks5qMaregaJpZM4I14vI>
>> .
>>
>
>